openedx_learning/apps/authoring/backup_restore/serializers.py

```diff
@@ -31,12 +47,56 @@ def validate(self, attrs):
         return attrs
 
 
-class ComponentVersionSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+class ComponentVersionSerializer(EntityVersionSerializer):  # pylint: disable=abstract-method
     """
     Serializer for component versions.
     """
-    title = serializers.CharField(required=True)
-    entity_key = serializers.CharField(required=True)
-    created = serializers.DateTimeField(required=True)
-    created_by = serializers.CharField(required=True, allow_null=True)
     content_to_replace = serializers.DictField(child=serializers.CharField(), required=True)
+
+
+class ContainerSerializer(EntitySerializer):  # pylint: disable=abstract-method
+    """
+    Serializer for containers.
+    """
+    container = serializers.DictField(child=serializers.DictField(), required=True)
+
+    def validate(self, attrs):
+        """
+        Custom validation logic:
+        parse the entity_key into (component_type, local_key).
+        """
+        try:
+            container = attrs["container"]
+            container_type = list(container.keys())[0]
+            if container_type not in ("section", "subsection", "unit"):
+                raise ValueError(f"Invalid container type: {container_type}")
+            attrs["container_type"] = container_type
+            attrs.pop("container")  # Remove the container field after processing
+        except ValueError as exc:
+            raise serializers.ValidationError({"key": str(exc)})
+        return attrs
+
+
+class ContainerVersionSerializer(EntityVersionSerializer):  # pylint: disable=abstract-method
+    """
+    Serializer for container versions.
+    """
+    container = serializers.DictField(child=serializers.ListField(child=serializers.CharField()), required=True)
+
+    def validate(self, attrs):
+        """
+        Custom validation logic:
+        parse the entity_key into (component_type, local_key).
+        """
+        try:
+            container = attrs["container"]
+            if "children" not in container:
+                raise ValueError("Missing 'children' in container")
+            children = container["children"]
+            if not isinstance(children, list):
+                raise ValueError("'children' must be a list")
+            attrs["children"] = children
+            attrs.pop("container")  # Remove the container field after processing
+        except ValueError as exc:
+            raise serializers.ValidationError({"key": str(exc)})
```

Comment:
Oh, I see, we're aggregating errors and re-raising. The nice thing about using separate `validate_<field>` methods is that it can surface multiple errors at the same time. This pattern you have here will short-circuit on the first error and hide the other ones.